portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -30,77 +31,65 @@
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
 import com.google.gson.Gson;
-import com.google.appengine.api.datastore.Key;
-import com.google.appengine.api.datastore.KeyFactory;
 
 /** Servlet that encapsulates some data from training exercises. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+
   /* Do Get function to fetch and list the todo list items onto the home page*/
   @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException 
+  {
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
 
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
     // Private class which is supposed to act like a Comment class.
     List<Comment> comments = new ArrayList<>();
-    for (Entity entity : results.asIterable()) {
+    for (Entity entity : results.asIterable(FetchOptions.Builder.withLimit(Integer.parseInt(getParameter(request, "limit", "")))) ) {
```

Comment:
parseInt can throw an error if the user hasn't actually given you a number, you should assign it to a variable inside a try catch.

Reply:
Fixed, now it works!